src/wasm/WasmModule.cpp

```diff
@@ -562,139 +644,19 @@ int WasmModule::awaitPthreadCall(const faabric::Message* msg, int pthreadPtr)
 
     int returnValue = sch.awaitThreadResult(callId);
 
-    pthreadPtrsToChainedCalls.erase(pthreadPtr);
-
-    return returnValue;
-}
-
-void WasmModule::setUpOpenMPMergeRegions(
-  const faabric::Message& msg,
-  std::shared_ptr<threads::Level> ompLevel)
-{
-    std::string snapshotKey = msg.snapshotkey();
-
-    if (snapshotKey.empty()) {
-        return;
-    }
-
-    // Create ordered list of offsets
-    std::vector<uint32_t> sortedOffsets(ompLevel->sharedVarOffsets.get(),
-                                        ompLevel->sharedVarOffsets.get() +
-                                          ompLevel->nSharedVarOffsets);
-
-    std::sort(sortedOffsets.begin(), sortedOffsets.end());
+    // Decrease the count and see if we're the last one
+    int prevValue = queuedPthreadCallsCount.fetch_sub(1);
```

Comment:
Set memory order

Reply:
On second thoughts, this operation is now only single threaded so this synchronisation isn't needed.